controllers/configurationpolicy_controller.go

```diff
@@ -202,49 +205,45 @@ func (r *ConfigurationPolicyReconciler) PeriodicallyExecConfigPolicies(
 			skipLoop = true
 		}
 
-		if !skipLoop && (discoveryErr == nil || cleanupImmediately) {
+		if !skipLoop {
```

Comment:
If the API server is down of the hosted cluster in hosted mode, we still want to clean up the finalizers on the management cluster.
```suggestion
		if !skipLoop || cleanupImmediately {
```

Reply:
Right